.github/workflows/test-deploy.yml

```diff
@@ -102,47 +102,29 @@ jobs:
           registry-url: https://registry.npmjs.org
       - run: npm install
       - run:
-          npm ${{ inputs.release && 'publish' || 'publish --dry-run' }}
+          npm ${{ inputs.release && 'publish' || 'publish --dry-run' }} || ${{
```

Comment:
why not just use conditional workflow steps?

Reply:
Agreed, it would be much clearer for any post analysis.

Reply:
But we'd need to use up some of our newline quota :smile:

Reply:
Note that the extra `|| ${{ inputs.release && 'true' || 'false' }}` was immediately reverted with #1215; I added it to make the (non-dry) publish process idempotent, because trying to publish an existing version would fail.

Reply:
### Ternary operator
```yaml
- run: npm ${{ inputs.release && 'publish' || 'publish --dry-run' }}
  env:
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
### Conditional steps
```yaml
- if: inputs.release == true
  run: npm publish
  env:
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
- if: inputs.release != true
  run: npm publish --dry-run
  env:
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Reply:
Happy to deplete our newline quota if you feel it's necessary 😄

Reply:
WTB this as a future required change for the next person to edit said lines.